core/tests/protractor_utils/AdminPage.js

```diff
@@ -152,6 +159,37 @@ var AdminPage = function() {
       'Could not set role successfully');
     return true;
   };
+
+  this.viewRolesbyRole = function(role) {
+    waitFor.visibilityOf(roleDropdown,
+      'View role dropdown taking too long to be visible');
+    roleDropdown.sendKeys('By Role');
+
+    roleValueOption.click();
+    roleValueOption.sendKeys(role);
+
+    viewRoleButton.click();
+    return true;
+  };
+
+  this.viewRolesbyUsername = function(username) {
+    waitFor.visibilityOf(roleDropdown,
+      'View role dropdown taking too long to be visible');
+    roleDropdown.sendKeys('By Username');
+
+    roleUsernameOption.click();
+    roleUsernameOption.sendKeys(username);
+
+    viewRoleButton.click();
+    return true;
+  };
+
+  this.expectNumberOfUsers = function(expectedCount) {
```

Comment:
Can you also verify that the correct users are returned? Something like expectUsersToBe([list of users to verify against])

Reply:
Done